algorithms/src/polycommit/sonic_pc/polynomial.rs

```diff
@@ -249,6 +250,7 @@ impl<'a, F: PrimeField> LabeledPolynomialWithBasis<'a, F> {
                     Lagrange { evaluations } => {
                         let domain = evaluations.domain().size();
                         if let Some(e) = lagrange_polys.get_mut(&domain) {
+                            ensure!(e.len() == evaluations.evaluations.len());
```

Comment:
Hmm now that you mention, I think that zip only holds under a specific condition: e2674ba50
However, plot twist, this is dead code which we never use. We also needlessly clone all polys in this setup. Removing it all here: https://github.com/AleoHQ/snarkVM/pull/2207

In the meantime I'll continue the search for remaining panics to avoid.